packages/core-p2p/src/network-monitor.ts

```diff
@@ -464,12 +455,12 @@ export class NetworkMonitor implements Contracts.P2P.NetworkMonitor {
         }
         // Save any downloaded chunks that are higher than a failed chunk for later reuse.
         for (i++; i < chunksToDownload; i++) {
-            if (
-                downloadResults[i] !== undefined &&
```

Comment:
In the refactored implementation you do not check for `downloadResults[i]` to be defined, is it okay still ?